### PR TITLE
ops #102610: Responsive DeviceInfo

### PR DESCRIPTION
**Pull request by ErlendAHall**
## What's new
- Added a proxying class that refreshes DeviceInfo on viewport dimension changes.
- Added hook useDeviceInfo for use in react components. It exports an updated deviceInfo + booleans for the 3 different deviceTypes.
- Replaced usages of useIsMobile and other queries to DeviceInfo.deviceType

This should streamline development more as `deviceInfo.orientation`, `deviceInfo.deviceType` and `deviceInfo.userInput` now change automatically without needing to refresh the page.

Note1: There was previously a mishmash of isMobile checks. Sometimes isMobile meant only "mobile" and other times both "mobile" and "tablet". This is now intentionally mobile only.
Note2: The Sidesheets still do not respond to changes. An update in Echo Components might be necessary.

### CODE REVIEW

File: src/components/SideSheets/library/collection/CollectionList.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ interface CollectionListProps {
 
 function CollectionList(props: CollectionListProps) {
     const collections = props.collections;
+    useDeviceInfo();
```
Comment (Tobzor) [inline src/components/SideSheets/library/collection/CollectionList.tsx:20]: Should this be used like that?

File: src/components/common/info/InfoBlock.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React, { CSSProperties } from 'react';
 
 import styles from './Info.module.css';
 import { classnames } from '@equinor/echo-components';
+import { useDeviceInfo } from '@components';
```
Comment (Tobzor) [inline src/components/common/info/InfoBlock.tsx:5]: same alias -- should probably be relative import

File: src/components/controlbar/ControlBar.tsx [hunk 3]
```diff
@@ -6,10 +6,9 @@ import {
     SideSheet,
     SideSheetOrientation,
 } from '@equinor/echo-components';
-import { deviceInfo } from '@equinor/echo-framework';
 
 import { devToggles, mapsStorage } from '@utils';
-import { useObservable, useMapsContext } from '@components';
+import { useObservable, useMapsContext, useDeviceInfo } from '@components';
```
Comment (Tobzor) [inline src/components/controlbar/ControlBar.tsx:11]: might need to have relative imports here -- this however might work but it all now depends on the order of exports in components/index file

File: src/components/controlbar/mapControls/MapControls.tsx [hunk 4]
```diff
@@ -1,7 +1,6 @@
 import React from 'react';
-import { deviceInfo } from '@equinor/echo-framework';
 
-import { MapIndicator, useMapsContext } from '@components';
+import { MapIndicator, useMapsContext, useDeviceInfo } from '@components';
```
Comment (Tobzor) [inline src/components/controlbar/mapControls/MapControls.tsx:3]: relative import might be needed eventually

File: src/components/mapIndicator/MapIndicator.tsx [hunk 5]
```diff
@@ -1,13 +1,11 @@
 import React, { useEffect, useMemo, useRef, useState } from 'react';
 
 import { Button, Divider, Icon, Popover } from '@equinor/eds-core-react';
-import { deviceInfo } from '@equinor/echo-framework';
 
+import { useDeviceInfo, useMapsContext, useObservable } from '../hooks';
```
Comment (Tobzor) [inline src/components/mapIndicator/MapIndicator.tsx:5]: nice, this is correct import :D

File: src/components/SideSheets/library/collection/CollectionList.tsx (hunk 1 above)
Comment (ErlendAHall) [reply src/components/SideSheets/library/collection/CollectionList.tsx:20]: No, I have removed it. Must be testing leftover

**Review by Tobzor — APPROVED** [review]

**Pull request merged**